### ISSUE TITLE
Cannot start virtual camera with filter enabled

### ISSUE THREAD

**Issue by knw257** [exeldro/obs-virtual-cam-filter#1]
Description: When the Virtual Camera filter is enabled on a scene, and the OBS virtual camera has not been started this run, the camera will fail to start. In the application log with verbose logging enabled, the only relevant log entry appears to be "16:28:50.503: starting virtual-output failed"

Steps to reproduce:
1. Install virtual camera plugin
2. Start OBS
3. Apply Virtual Camera filter to a single scene
4. Restart OBS
5. Attempt to start virtual camera

Expected behavior: Virtual camera would start with filtered scene being output regardless of which scene is actively selected
Actual behavior: OBS virtual camera fails to start, log only contains entry "starting virtual-output failed", applications trying to use OBS virtual camera only show image indicating OBS camera is offline

Workaround:
1. After installing plugin and adding filter to a scene, deactivate the filter
2. Start the OBS virtual camera
3. Activate the filter
4. Stop OBS virtual camera

This appears to get the plugin to work as desired

System details:
OS: 
> Windows 10 Pro
> Version	20H2
> Installed on	‎8/‎25/‎2020
> OS build	19042.685
> Experience	Windows Feature Experience Pack 120.2212.551.0

OBS: 26.1.0 64-bit

Log file with options --verbose --startvirtualcam attached
[2020-12-30 17-08-24.txt](https://github.com/exeldro/obs-virtual-cam-filter/files/5755863/2020-12-30.17-08-24.txt)

**Comment by Zx-EvM** [comment]
Don't use the start/stop virtual camera (default buttons). When you add the filter you just need to switch off/on it with toggle the eye icon.

**Comment by carpet92** [comment]
@knw257 Don't use the start/stop virtual camera (default buttons). When you add the filter you just need to switch off/on it with toggle the eye icon.

**Comment by carpet92** [comment]
@knw257 there is another issue with applying the changes to virtual camera scene (adding the images or other effects) that does not shares for applications: https://github.com/exeldro/obs-virtual-cam-filter/issues/2

**Comment by knw257** [comment]
> @knw257 Don't use the start/stop virtual camera (default buttons). When you add the filter you just need to switch off/on it with toggle the eye icon.

Fair enough, but when starting OBS with the filter enabled, nothing is being sent to the OBS VCam. It seems the filter has to be flipped on after OBS startup in order to work properly

**Comment by carpet92** [comment]
@knw257 I demonstrating how to fix it without use the filter but with separation Virtual Camera to another OBS instance in my video in the issue #2 since 03:22

**Comment by carpet92** [comment]
@knw257 let me know please if this method will works for you too.

**Comment by carpet92** [comment]
The following appears to get the plugin to work as desired:

1. Run OBS
2. Add new SCENE
3. Add the webcam fullscreen as source in the SCENE
4. Add plugin to source and for the SCENE
5. Toggle eye button of the plugin firstly for source then for SCENE
6. Toggle eye button of the plugin firstly for SCENE then for source (yes, the same but in reverse order)
7. Repeat step 6 on each time you start OBS

Now all added images or other filters will appears to show for all applications that uses OBS Virtual Camera. What a buggy magic with that plugin.

P.S. but sometimes when you exit the OBS you will get crash message.

**Comment by artsra** [comment]
This plugin definitely crashes a lot, or causes OBS to crash when I click the eye-icon to make it visible.

**Comment by MythicalRollers** [comment]
Same as artsra. OBS 28.0.3
Virtual Cam Filter 0.0.6
If I turn on the filter, OBS crashes.
